docs/interfaces.md

```diff
@@ -12,184 +12,349 @@ Slots may support multiple connections to plugs.  For example the OS snap
 exposes the ``network`` slot and all applications that can talk over the
 network connect their plugs there.
 
-## Supported Interfaces - Basic
+The availability of an interface depends on whether snapd is running on a
+classic (eg, traditional desktop or server) or on a native system. Interfaces
+may also be implicit to the OS snap or implemented only via snaps providing the
+slot.
 
-### network
+## Native vs classic interfaces
```

Comment:
Needs some rewording. There's no "classic interface" and "native interface" in snapd. There's the classic Ubuntu system, and there's Ubuntu Core. The interface set on both of those is the same.